app/controllers/actions/AuthenticateGetOptionalArrivalForWriteProvider.scala

```diff
@@ -57,11 +57,11 @@ class AuthenticateGetOptionalArrivalForWriteAction(
     request.body match {
       case body: NodeSeq =>
         XmlMessageParser.mrnR(body) match {
-          case None =>
-            Logger.warn("Invalid mrn specified in request")
-            Future.successful(BadRequest("Invalid mrn specified in request"))
+          case Left(error) =>
+            Logger.warn(s"Failed to retrieve MovementReferenceNumber with error: $error")
```

Comment:
I think it's worth properly showing the error. `Logger.error(msg, error)` or I am not sure if you can pass through an error object to the `Logger.warn` method, but we would need to, because in here we are throwing away the stack trace right now.